airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

```diff
@@ -97,7 +100,15 @@ def get_granted_scopes(self, authenticator):
         try:
             access_token = authenticator.get_access_token()
             url = f"https://api.hubapi.com/oauth/v1/access-tokens/{access_token}"
-            response = requests.get(url=url)
+            error_resolution = ErrorResolution(ResponseAction.RETRY, FailureType.system_error, "Internal error attempting to get scopes.")
+            http_client = HttpClient(
+                name="get hubspot granted scopes client",
+                logger=self.logger,
+                error_handler=HttpStatusErrorHandler(logger=self.logger, error_mapping={500: error_resolution, 502: error_resolution}),
+                backoff_strategy=DefaultBackoffStrategy(),
+            )
+            prepared_request = http_client._create_prepared_request(http_method="get", url=url)
+            response = http_client._send_with_retry(prepared_request, request_kwargs={})
```

Comment:
This is using private methods from HttpClient. Why not use `http_client.send_request` instead?

Reply:
done